src/app/SIPUserAgents/SIPServerUserAgent.cs

```diff
@@ -489,11 +489,11 @@ private Task<SocketError> ByeServerFinalResponseReceived(SIPEndPoint localSIPEnd
             }
         }
 
-        private void UASTransactionCancelled(SIPTransaction sipTransaction)
+        private void UASTransactionCancelled(SIPTransaction sipTransaction, SIPRequest sipCancelRequest)
         {
             logger.LogDebug("SIPServerUserAgent got cancellation request.");
             m_isCancelled = true;
-            CallCancelled?.Invoke(this);
+            CallCancelled?.Invoke(this, sipCancelRequest?.Header.Reason, sipCancelRequest?.Header.UnknownHeaders.ToArray());

```

Comment:
Is there any reason not to pass sipCancelRequest instead of parts of it? THe full request would seem more flexible for future use.

Reply:
SipCancelRequest can't be mocked, it has no interface.

Reply:
I changed this to pass the full SIPRequest as suggested.